api/src/pcapi/routes/backoffice/collective_offers/collective_offers_blueprint.py

```diff
@@ -37,100 +41,265 @@
-            educational_models.CollectiveOffer.validation,
-            educational_models.CollectiveOffer.formats,
-            educational_models.CollectiveOffer.authorId,
-        ),
-        sa.orm.joinedload(educational_models.CollectiveOffer.collectiveStock).load_only(
-            educational_models.CollectiveStock.beginningDatetime
-        ),
-        sa.orm.joinedload(educational_models.CollectiveOffer.venue).load_only(
-            offerers_models.Venue.managingOffererId, offerers_models.Venue.name, offerers_models.Venue.publicName
+aliased_stock = sa.orm.aliased(educational_models.CollectiveStock)
+
+SEARCH_FIELD_TO_PYTHON = {
+    "FORMATS": {
+        "field": "formats",
+        "column": educational_models.CollectiveOffer.formats,
+    },
+    "INSTITUTION": {
+        "field": "institution",
+        "column": educational_models.CollectiveOffer.institutionId,
+    },
+    "CREATION_DATE": {
+        "field": "date",
+        "column": educational_models.CollectiveOffer.dateCreated,
+        "special": partial(date_utils.date_to_localized_datetime, time_=datetime.datetime.min.time()),
+    },
+    "DEPARTMENT": {
+        "field": "department",
+        "column": offerers_models.Venue.departementCode,
+        "inner_join": "venue",
+    },
+    "REGION": {
+        "field": "region",
+        "column": offerers_models.Venue.departementCode,
+        "inner_join": "venue",
+        "special": regions_utils.get_department_codes_for_regions,
+    },
+    "EVENT_DATE": {
+        "field": "date",
+        "column": aliased_stock.beginningDatetime,
+        "special": partial(date_utils.date_to_localized_datetime, time_=datetime.datetime.min.time()),
+        "inner_join": "stock",
+    },
+    "BOOKING_LIMIT_DATE": {
+        "field": "date",
+        "column": aliased_stock.bookingLimitDatetime,
+        "special": partial(date_utils.date_to_localized_datetime, time_=datetime.datetime.min.time()),
+        "inner_join": "stock",
+    },
+    "ID": {
+        "field": "integer",
+        "column": educational_models.CollectiveOffer.id,
+    },
+    "NAME": {
+        "field": "string",
+        "column": educational_models.CollectiveOffer.name,
+    },
+    "OFFERER": {
+        "field": "offerer",
+        "column": offerers_models.Venue.managingOffererId,
+        "inner_join": "venue",
+    },
+    "STATUS": {
+        "field": "status",
+        "column": educational_models.CollectiveOffer.status,
+    },
+    "VENUE": {
+        "field": "venue",
+        "column": educational_models.CollectiveOffer.venueId,
+    },
+    "VALIDATION": {"field": "validation", "column": educational_models.CollectiveOffer.validation},
+}
+
+JOIN_DICT: dict[str, list[dict[str, typing.Any]]] = {
+    "stock": [
+        {
+            "name": "stock",
+            "args": (
+                aliased_stock,
+                aliased_stock.collectiveOfferId == educational_models.CollectiveOffer.id,
+            ),
```

Comment:
[mineur/style]
```suggestion
            "args": (
                aliased_stock,
                educational_models.CollectiveOffer.collectiveStock,
            ),
```

Fonctionnellement il n'y a pas de différence avec ce qu tu as fait mais je trouve ça plus lisible